### PR TITLE
267: Add http json functionality

### PR DESCRIPTION
**Pull request by paulfgrant01**
based on [Issue 267](https://github.com/helmfile/vals/issues/267)
This provider will retrieve a single key from a JSON structure hosted by an HTTP frontend. The signature will require both the domain and the xpath e.g. 

ref+httpjson://&lt;DOMAIN&gt;/&lt;PATH&gt;?[insecure=false&floatAsInt=false]#/&lt;XPATH&gt;

### CODE REVIEW

**Comment by yxxhero** [comment]
@paulfgrant01 please fix lint issue.

File: pkg/providers/httpjson/httpjson.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
+package httpjson
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	// nolint
```
Comment (yxxhero) [inline pkg/providers/httpjson/httpjson.go]: nolint?

File: vals.go [hunk 2]
```diff
@@ -367,6 +372,7 @@ func (r *Runtime) prepare() (*expansion.ExpandRegexMatch, error) {
 				}
 
 				return str, nil
+
```
Comment (yxxhero) [inline vals.go]: extra empty lint?

File: pkg/providers/httpjson/httpjson.go (hunk 1 above)
Comment (paulfgrant01) [reply pkg/providers/httpjson/httpjson.go]: Removed in favor of:

gci write -s standard -s default -s "prefix(github.com/helmfile/vals)" pkg/providers/httpjson/httpjson.go

File: vals.go (hunk 2 above)
Comment (paulfgrant01) [reply vals.go]: Removed this line 👍

File: pkg/providers/httpjson/httpjson.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+	// Initialize docs map to store the json object for use multiple times
+	if len(p.docs) == 0 {
+		p.docs = make(map[string]*jsonquery.Node)
+	}
+
+	return p
+}
+
+func GetXpathFromUri(uri string) (xpath string, err error) {
```
Comment (yxxhero) [inline pkg/providers/httpjson/httpjson.go]: what will happen if uri is empty? and add unittest for function as possible as you can. thanks so much.

Comment (paulfgrant01) [reply pkg/providers/httpjson/httpjson.go]: Added unit testing 🙏

File: vals_httpjson_test.go [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+			"insecure":   "true",
+		},
+		"inline": map[string]interface{}{
+			"value": inlineValue,
+		},
+	}
+	return config2.Map(config)
+}
+
+// nolint
```
Comment (yxxhero) [inline vals_httpjson_test.go]: why nolint? I can help you to solve lint issue.

Comment (yxxhero) [reply vals_httpjson_test.go]: remove this. then I will merge this PR.

Comment (paulfgrant01) [reply vals_httpjson_test.go]: Removed nolint

File: pkg/providers/httpjson/httpjson.go (hunk 3 above)
Comment (paulfgrant01) [reply pkg/providers/httpjson/httpjson.go]: Thank you 🙏

**Pull request merged**